pkg/oc/cli/cmd/set/env.go

```diff
@@ -198,6 +203,10 @@ func (o *EnvOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, args []s
 		return kcmdutil.UsageErrorf(o.Cmd, "--list and --output may not be specified together")
 	}
 
+	if len(o.Keys) > 0 && len(o.From) < 1 {
+		return kcmdutil.UsageError(cmd, "when specifying --keys, a configmap or secret must be provided with --from")
+	}
```

Comment:
Mind moving this (as well as the above `if` statement) to a new `Validate()` method?
This would keep the create, validate, run pattern that we're trying to enforce across all commands